src/Notification/Notification.js

```diff
@@ -0,0 +1,351 @@
+		*
+		* @type {String}
+		* @default ''
+		* @public
+		*/
+		content: '',
+
+		/**
+		* The duration, in miliseconds, that the notification takes to animate on and off screen.
+		* Setting to zero (0) forces the transition on/off screen to be instant. However, if your
+		* desire is to have immediate transitions, it's more efficient to simply toggle
+		* [animate]{@link module:moonstone/Notification~Notification#animate}.
+		*
+		* @type {Number}
+		* @default 400
+		* @public
+		*/
+		showHideDuration: 400,
+
+		/**
+		* If `true`, {@glossary Spotlight} (focus) cannot leave the area of the notification unless
+		* the notification is explicitly closed; if `false`, spotlight may be moved anywhere within
+		* the viewport.
+		*
+		* @type {Boolean}
+		* @default true
+		* @public
+		*/
+		spotlightModal: true
+	},
+
+	/**
+	* @private
+	*/
+	components: [
+		{name: 'message', kind: BodyText},
+		{name: 'client', kind: Control, classes: 'enyo-fill client moon-hspacing'}
+	],
+
+	/**
+	* @private
+	*/
+	bindings: [
+		{from: 'content', to: '$.message.content'},
+		{from: 'allowHtml', to: '$.message.allowHtml'}
+	],
+
+	/**
+	* @private
+	*/
+	create: function () {
+		Popup.prototype.create.apply(this, arguments);
+		this.animateChanged();
+	},
+
+	/**
+	* @private
+	*/
+	render: function () {
+		this._initialized = true;
+		Popup.prototype.render.apply(this, arguments);
+	},
+
+	/**
+	* @private
+	*/
+	showHideDurationChanged: function () {
+		var dur = (this.animate && this.showHideDuration) ? this.showHideDuration : 0;
+		this.set('showingDuration', dur);
+		this.set('hidingDuration', dur);
+		this.applyStyle('-webkit-transition-duration', dur + 'ms');
+		this.applyStyle('-moz-transition-duration', dur + 'ms');
+		this.applyStyle('transition-duration', dur + 'ms');
+	},
+
+	/**
+	* @private
+	*/
+	animateChanged: function () {
+		this.addRemoveClass('animate', this.animate);
+		this.showHideDurationChanged();
+	},
+
+	/**
+	* @private
+	*/
+	showingTransitioningChanged: function () {
+		if (this.showingTransitioning) {
+			// if we are currently animating the hide transition, release
+			// the events captured when popup was initially shown
+			if (this.captureEvents) {
+				this.release();
+			}
+		}
+
+		if (!this.showing) {
+			Spotlight.unspot();
+		}
+	},
+
+	/**
+	* @private
+	*/
+	shown: function (sender, ev) {
+		this.activator = Spotlight.getCurrent();
+
+		var current = this.activator;
+		if (!current || !current.isDescendantOf(this)) {
+			if (Spotlight.isSpottable(this)) {
+				Spotlight.setPointerMode(false);
+				Spotlight.spot(this);
+			}
+			// If we're not spottable, just unspot whatever was previously spotted
+			else {
+				Spotlight.unspot();
+			}
+		}
+
+		if (this.allowBackKey) {
+			this.pushBackHistory();
+		}
+	},
+
+	/**
+	* @private
+	*/
+	hidden: function (sender, ev) {
+		this.respotActivator();
+
+		if (this._initialized && this.allowBackKey && !EnyoHistory.isProcessing()) {
```

Comment:
What do you think about checking for something like `this.hasNode()` here, instead of maintaining an internal property? Would also allow us to get rid of the `render` override.